libbeat/outputs/logstash/async.go

```diff
@@ -135,6 +141,13 @@ func (c *asyncClient) Close() error {
 }
 
 func (c *asyncClient) Publish(_ context.Context, batch publisher.Batch) error {
+	c.connMu.RLock()
```

Comment:
Locking here is a potential deadlock because there's a call to `Close` below -- I actually preferred the lock in `sendEvents` for that reason :-) But if you do it in `Publish` you'll need to manage the unlock manually instead of with `defer`.

Reply:
Thanks, I will step back and rethink this a bit. I think it is clearer to use a normal mutex and lock in sendEvents instead.

Reply:
Changed to use a simple mutex like before. Can you check if that is what you had in mind?